components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/endpoint/impl/impl/ResourceRegistrationApiServiceImpl.java

```diff
@@ -0,0 +1,247 @@
+            handleErrorResponse(e, log);
+        } catch (UMAClientException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Client error when registering resource in resource server.", e);
+            }
+            handleErrorResponse(e, log);
+        } catch (Throwable throwable) {
+            log.error("Internal server error occurred. ", throwable);
+        }
+        return null;
+    }
+
+    /**
+     * Retrieve the resource of the given resourceId
+     *
+     * @param resourceId Resourceid of the resource which need to get retrieved
+     * @return Response with the retrieved resource/ retrieval status
+     */
+
+    @Override
+    public Response getResource(String resourceId) {
+
+        Response response = null;
+
+        try {
+            if (!isResourceId(resourceId)) {
+                try {
+                    Resource resourceRegistration = ResourceUtils.getResourceService().getResourceById
+                            (resourceId);
+                    ReadResourceDTO readResourceDTO = ResourceUtils.readResponse(resourceRegistration);
+                    return Response.status(Response.Status.OK).entity(readResourceDTO).build();
+                } catch (UMAServiceException e) {
+                    handleErrorResponse(e, log);
+                } catch (Throwable throwable) {
+                    log.error("Internal server error occurred. ", throwable);
+                }
+
+            }
+        } catch (UMAClientException e) {
+            handleErrorResponse(e, log);
+            if (log.isDebugEnabled()) {
+                log.debug("Client error when retrieving resource from the resource server.", e);
+            }
+        }
+        return response;
+    }
+
+    /**
+     * Retrieve the available resourceId list
+     *
+     * @param resourceOwnerId
+     * @return Response with the retrieved resourceId's/ retrieval status
+     */
+    @Override
+    public Response getResourceIds(String resourceOwnerId) {
+
+        // For testing purpose current resource owner id is taken to validate pat access token
+        resourceOwnerId = "123";
+        try {
+            List<String> resourceRegistration = ResourceUtils.getResourceService().getResourceIds(resourceOwnerId);
+            Response response = Response.ok().entity(resourceRegistration).build();
+            return response;
+        } catch (UMAClientException e) {
+            handleErrorResponse(e, log);
+            log.error("Invalid request.Request with valid resource Id to update the resource. ", e);
+            //return Response.status(status).entity(errorDTO).build();
+
+        } catch (UMAException e) {
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+        return null;
+    }
+
+    /**
+     * Update a resource
+     *
+     * @param updatedResource details of the resource to be updated
+     * @param resourceId      ID of the resource to be updated
+     * @return
```

Comment:
You need to fill all the details when adding a doc comment. Please fix all the occurrences.